nifi-commons/nifi-utils/src/test/java/org/apache/nifi/processor/TestFormatUtils.java

```diff
@@ -103,4 +111,41 @@ public void testFormatDataSize() {
 
         assertEquals(String.format("181%s9 TB", decimalFormatSymbols.getDecimalSeparator()), FormatUtils.formatDataSize(200_000_000_000_000d));
     }
+
+    @Test
+    public void testParseInstant() throws Exception {
+        TimeZone current = TimeZone.getDefault();
+        TimeZone.setDefault(TimeZone.getTimeZone("Europe/Kiev"));
```

Comment:
I'm testing here different time zone combination, there are explicitly passed Europe/Warsaw and system default (set to Europe/Kiev), what will give us testing with more combinations? Java time API handles in the same way time zones greater and less than GMT.

Reply:
Recently we had some issues that occurred only in UTC+ or UTC- time zones.
I see there is no high chance for this here but as a rule of thumb, I would always write tests with UTC, UTC+ and UTC- zones.
In this case, it would mean 3 x 3 tests (3 time zones for `TimeZone.setDefault()` and input data from 3 time zones). The input data can easily be added (2 more lines below) and the test can be converted to a parameterized one to use different `TimeZone.setDefault()` settings.

Reply:
Wow I wonder what kind of bug it was that it was related to this -/+ things? At the end subtraction is just an addition of negative number ;)  I've added all those combinations.